lib/tests/formulae_detect.rb

```diff
@@ -75,7 +75,13 @@ def detect_formulae!(args:)
           else
             test git, "-C", repository, "fetch", "origin", "+#{github_ref}" unless tap.official?
             origin_ref = "origin/#{github_ref.gsub(%r{^refs/heads/}, "")}"
-            diff_end_sha1 = diff_start_sha1 = github_sha
+            diff_end_sha1 = github_sha
+            diff_start_sha1 = if (event_name = ENV.fetch("GITHUB_EVENT_NAME", nil)).present? &&
+                                 event_name == "merge_group"
+              rev_parse("#{github_sha}^")
```

Comment:
I don't think this is right if the PR contains multiple commits. I think you want do a comparison to `origin_ref` instead to trigger a `git diff origin/master`. `git merge-base origin/master` may be useful, too/instead.

Reply:
I tried it out locally and it seemed right to me. Or are you referring to the case where a merge group handles multiple PRs at a time? Because I'll admit to having no idea how those even work.

Do you mind giving this a try to give us a better idea of the tweaks needed?